app/controllers/catalog_controller.rb

```diff
@@ -1,13 +1,12 @@
 # frozen_string_literal: true
 class CatalogController < ApplicationController
-
   include BlacklightRangeLimit::ControllerOverride
 
   include Blacklight::Catalog
 
+  # rubocop:disable Security/Eval
```

Comment:
This is a piece of the work I am unsure of. We are getting it from ourselves so I want to say it's safe.

Reply:
I think you can undo the eval with this:
    # Parsing the JSON response
    rsp = JSON.parse(http_response.body)

    # Accessing the desired data from the parsed JSON
    params[:id] = rsp['response']['docs'][0]['id'] if rsp['response']['docs'] && !rsp['response']['docs'].empty?